server/src/storage/customers.ts

```diff
@@ -30,10 +34,44 @@ const getCustomer = async (customerId: number): Promise<CustomerResponse> => {
       value: 'ongoing',
     },
   ]);
+  return commits.length;
+};
+
+const getCustomer = async (customerId: number): Promise<CustomerResponse> => {
+  const customer: CustomerDatastoreReponse = await get(
+    CUSTOMER_KIND,
+    customerId
+  );
+  const numOngoingCommits = await getNumOngoingCommits(customerId);
+  return {
+    ...customer,
+    numOngoingCommits,
+  };
+};
+
+/**
+ * Gets a customer with the specified gpayId.
+ * Returns a CustomerResponse object when such a customer exits.
+ * Returns null otherwise.
+ * @param gpayId The gpay id of the customer to retrieve
+ */
+const getCustomerWithGpayId = async (
+  gpayId: string
+): Promise<CustomerResponse | null> => {
+  const [customer]: CustomerDatastoreReponse[] = await getAll(CUSTOMER_KIND, [
+    {
+      property: 'gpayId',
```

Comment:
Can datastore enforce uniqueness/index on gpayId?

Reply:
@ifhuang Datastore is already indexed on gpayId! And uniqueness-wise datastore does not provide it so I'm the one implementing it in my `addCustomer` which uses a transaction to first ensure the same gpayId does not already exist.

Reply:
SGTM